### ISSUE TITLE
BigNum - elapsed time

### ISSUE THREAD

**Issue by Pako2** [herrnst/script.xbmc.lcdproc#62]
I recently became a user of the XBMC LCDproc add-on (BTW - thank you for your great work !).
However, I have one problem. When I listen to the radio, the display shows the current position of the currently playing program (according to the beginning obtained from the EPG).
But wrong. I have a 2x16 format display and the position is ALWAYS shown in MM:SS format. The number of hours elapsed is never displayed. **This is very confusing.**

I think you need to modify the infolabels.py and lcdproc.py modules:
```
#infolabels.py
#=============
    def GetPlayerTime(self):
        if self.PlayingLiveTV() or self.PlayingLiveRadio():
            #return self.GetInfoLabel("PVR.EpgEventElapsedTime")
            return self.GetInfoLabel("PVR.EpgEventElapsedTime(hh:mm:ss)")

        return self.GetInfoLabel("Player.Time")


#lcdproc.py
#==========
  def GetBigDigitTime(self, mode):
      ret = ""

      if self.m_InfoLabels.IsPlayerPlaying():
        if not (mode == LCD_MODE.LCD_MODE_SCREENSAVER and self.m_InfoLabels.IsPlayerPaused()):
          #ret = self.m_InfoLabels.GetPlayerTime()[-self.m_iBigDigits:]
          ret = self.m_InfoLabels.GetPlayerTime()[:self.m_iBigDigits]

      if ret == "": # no usable timestring, e.g. not playing anything
        strSysTime = self.m_InfoLabels.GetSystemTime()

        if self.m_iBigDigits >= 8: # return h:m:s
          ret = strSysTime
        elif self.m_iBigDigits >= 5: # return h:m when display too small
          ret = strSysTime[:5]

      return ret
```
Then it works properly (in my opinion), the position is displayed in the format HH:MM.
Unfortunately, I have no option to try it with another display (eg 4x20).

Note:
I originally posted this post in the Kodi forum, but there was no response.

**Comment by herrnst** [comment]
MM:SS is intentional for smaller displays since BigTime is mainly used for "large" time displays when playing music and such things.

**Comment by Pako2** [comment]
Thanks for the answer, but I can't understand it.
Is that what users really want?
If a radio show is longer than an hour (there are many of them in our country), then the information MM:SS is completely meaningless. And in that case, seconds don't matter so much.
So will it remain so confusing?
And would it be possible to add a new item to the settings, where the user would have the option to choose the second variant ("HH:MM" instead of "MM:SS")?
Otherwise I will be forced to make the adjustment myself after each upgrade ...
Thanks !

**Comment by herrnst** [comment]
As said, BigTime is mainly used for elapsed time display for music and maybe movies. Changing the format would have a direct effect also for these modes, and it wouldn't make much sense to have HH:MM shown for especially music, when thinking of three-to-four-minute tracks.

**Comment by Pako2** [comment]
Ah, that's an argument I understand. It didn't occur to me because I don't use Kodi to listen to songs.
But now I've tried. I connected a music disc and listened to some songs.
And I was surprised. Without me changing anything (so my proposed modification is valid), it works as you write! Thus, the elapsed time of a song is displayed in MM:SS format, while the elapsed time of a radio program is displayed in HH:MM format.
I also wanted to try BigNum while playing a video. But I couldn't, the display never switched to BigNum saver mode.
